agent/tcs/client/client.go

```diff
@@ -137,33 +137,40 @@ func (cs *clientServer) publishMetrics() {
 	// Publish metrics immediately after we connect and wait for ticks. This makes
 	// sure that there is no data loss when a scheduled metrics publishing fails
 	// due to a connection reset.
-	cs.publishMetricsOnce()
+	err := cs.publishMetricsOnce()
+	if err != nil && err != stats.EmptyMetricsError {
+		seelog.Warnf("Error getting instance metrics: %v", err)
```

Comment:
Can you rephrase the error message? The error is from the the `publishMetricsOnce()` method. We can't be sure here if its because of publish failing or if there was an error getting metrics.